File: 975-range-sum-of-bst/range-sum-of-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        if (root == null)
            return 0;

        Stack<TreeNode> stack = new Stack<>();
        int sum = 0;

        stack.push(root);

        while (stack.size() > 0) {
            TreeNode node = stack.pop();

            if (node.left != null) {
                stack.push(node.left);
            }
            if (node.val >= low && node.val <= high) {
                sum += node.val;
            }
            if (node.right != null) {
                stack.push(node.right);
            }
        }

        return sum;
    }
}

File: 975-range-sum-of-bst/range-sum-of-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */

/**
 * @param {TreeNode} root
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
var rangeSumBST = function (root, low, high) {
    return dfs(root, low, high);
};

const dfs = (node, low, high) => {
    if (!node) return 0;

    let sum = 0;

    if (node.val >= low && node.val <= high) {
        sum += node.val;
    }

    if (node.left) sum += dfs(node.left, low, high);
    if (node.right) sum += dfs(node.right, low, high);

    return sum;
};
